domain/notification/handler.go

```diff
@@ -26,25 +31,51 @@ type handler struct {
 	localFHIRClientFactory fhir.Factory
 	transferService        receiver.TransferService
 	registry               registry.OrganizationRegistry
+	vcr                    registry.VerifiableCredentialRegistry
 }
 
-func NewHandler(auth auth.Service, localFHIRClientFactory fhir.Factory, tranferReceiverService receiver.TransferService, registry registry.OrganizationRegistry) Handler {
+func NewHandler(
+	auth auth.Service,
+	localFHIRClientFactory fhir.Factory,
+	transferReceiverService receiver.TransferService,
+	registry registry.OrganizationRegistry,
+	vcr registry.VerifiableCredentialRegistry,
+) Handler {
 	return &handler{
 		auth:                   auth,
 		localFHIRClientFactory: localFHIRClientFactory,
-		transferService:        tranferReceiverService,
+		transferService:        transferReceiverService,
 		registry:               registry,
+		vcr:                    vcr,
 	}
 }
 
 // Handle handles an incoming notification about an updated Task for one of its customers.
 func (service *handler) Handle(ctx context.Context, notification Notification) error {
-	fhirServer, err := service.registry.GetCompoundServiceEndpoint(ctx, notification.SenderDID, "eOverdracht-sender", "fhir")
+	fhirServer, err := service.registry.GetCompoundServiceEndpoint(ctx, notification.SenderDID, transfer.SenderServiceName, "fhir")
 	if err != nil {
 		return fmt.Errorf("error while looking up custodian's FHIR server (did=%s): %w", notification.SenderDID, err)
 	}
 
-	accessToken, err := service.auth.RequestAccessToken(ctx, notification.CustomerDID, notification.SenderDID, "eOverdracht-sender", nil, nil)
+	taskPath := fmt.Sprintf("/Task/%s", notification.TaskID)
+
+	credentials, err := service.vcr.FindAuthorizationCredentials(
+		ctx,
+		&registry.VCRSearchParams{
+			PurposeOfUse: transfer.SenderServiceName,
```

Comment:
shouldn't we also filter on issuer? If more than 1 care organization issued a NutsAuthCredential to customerDID?